prow/git/v2/interactor.go

```diff
@@ -146,8 +148,43 @@ func (i *interactor) Clone(from string) error {
 	return nil
 }
 
+func (i *interactor) lock() error {
+	if !i.locking {
+		return nil
+	}
+	_, err := os.Stat(filepath.Join(i.dir, "lock"))
```

Comment:
Should we name this `prow-git-interactor-lock` or something other than just `lock`? I'm mainly worried about possible clashes with Git itself.